pkg/core/native/native_neo.go

```diff
@@ -200,6 +200,13 @@ func newNEO(cfg config.ProtocolConfiguration) *NEO {
 	md = newMethodAndPrice(n.unregisterCandidate, 1<<16, callflag.States)
 	n.AddMethod(md, desc)
 
+	desc = newDescriptor("onNEP17Payment", smartcontract.VoidType,
+		manifest.NewParameter("from", smartcontract.Hash160Type),
+		manifest.NewParameter("amount", smartcontract.IntegerType),
+		manifest.NewParameter("data", smartcontract.AnyType))
+	md = newMethodAndPrice(n.onNEP17Payment, 1<<15, callflag.States|callflag.AllowCall|callflag.AllowNotify, config.HFEchidna)
```

Comment:
We should probably schedule `registerCandidate` deprecation to the same hardfork, because if we plan to lower MaxBlockSystemFee at Echidna, then it won't be possible to register anyway.

Reply:
Step by step. Also, default `MaxBlockSystemFee` can differ from mainnet/testnet. And mainnet/testnet is the primary target here for now. We can roll Echidna out and tune `MaxBlockSystemFee` for mainnet/testnet after this release. Other network can use the old registration mechanism as well as `MaxBlockSystemFee`.